### PR TITLE
Integrate google internal changes.

### PR DESCRIPTION
**Pull request by pherl**

### CODE REVIEW

**Comment by pherl** [comment]
@xfxyjwf for review

**Comment by pherl** [comment]
I disabled the python2.6 support in this PR as the newly introduced json_format doesn't work under python2.6.

The tests should pass now. @xfxyjwf please take a look.

File: appveyor.yml [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-  - if %platform%==Win64 set generator=Visual Studio 12 Win64
-  - if %platform%==Win32 set vcplatform=Win32
-  - if %platform%==Win64 set vcplatform=x64
-
-build_script:
-  - CALL appveyor.bat
-
-skip_commits:
-  message: /.*\[skip appveyor\].*/
+# Only test one combination: "Visual Studio 12 + Win64 + Debug + DLL". We can
```
Comment (xfxyjwf) [inline appveyor.yml]: Could you revert this file? I believe the only change is about line-endings. We should probably exclude this file from the down-integrate script.

File: java/util/pom.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
   </parent>
   <groupId>com.google.protobuf</groupId>
   <artifactId>protobuf-java-util</artifactId>
-  <version>3.0.0-beta-1</version>
+  <version>3.0.0-alpha-4-pre</version>
```
Comment (xfxyjwf) [inline java/util/pom.xml]: Revert this file?

File: appveyor.yml (hunk 1 above)
Comment (pherl) [inline appveyor.yml]: Shall we change the git line ending settings to normalize them?

File: src/google/protobuf/util/default_value_objectwriter.h [hunk 3]
```diff
@@ -0,0 +1,218 @@
+#ifndef NET_PROTO2_UTIL_CONVERTER_INTERNAL_DEFAULT_VALUE_OBJECTWRITER_H_
```
Comment (xfxyjwf) [inline src/google/protobuf/util/default_value_objectwriter.h]: Revert this file? It should be in src/google/protobuf/util/internal and seems to be duplicated here.

File: appveyor.yml (hunk 1 above)
Comment (xfxyjwf) [inline appveyor.yml]: On Tue, Oct 6, 2015 at 1:29 PM, Jisi Liu <notifications@github.com> wrote:

> In appveyor.yml
> <https://github.com/google/protobuf/pull/849#discussion_r41317456>:
>
> > -  - ps: Start-FileDownload https://googlemock.googlecode.com/files/gmock-1.7.0.zip
> > -  - 7z x gmock-1.7.0.zip
> > -  - rename gmock-1.7.0 gmock
> > -
> > -before_build:
> > -  - if %platform%==Win32 set generator=Visual Studio 12
> > -  - if %platform%==Win64 set generator=Visual Studio 12 Win64
> > -  - if %platform%==Win32 set vcplatform=Win32
> > -  - if %platform%==Win64 set vcplatform=x64
> > -
> > -build_script:
> > -  - CALL appveyor.bat
> > -
> > -skip_commits:
> > -  message: /.*\[skip appveyor\].*/
> > +# Only test one combination: "Visual Studio 12 + Win64 + Debug + DLL". We can
>
> Shall we change the git line ending settings to normalize them?
>
Probably not. Currently all code under csharp directory are using DOS
line-endings and for Visual Studio users that makes more sense.

File: src/google/protobuf/util/default_value_objectwriter.h (hunk 3 above)
Comment (pherl) [inline src/google/protobuf/util/default_value_objectwriter.h]: Done

File: java/util/pom.xml (hunk 2 above)
Comment (pherl) [inline java/util/pom.xml]: Done

File: appveyor.yml (hunk 1 above)
Comment (pherl) [inline appveyor.yml]: Done

**Comment by xfxyjwf** [comment]
LGTM

**Pull request merged**